plugins/importexport/native/NativeImportExportPlugin.inc.php

```diff
@@ -13,408 +13,71 @@
-			case 'import':
-				AppLocale::requireComponents(LOCALE_COMPONENT_PKP_SUBMISSION);
-				$temporaryFileId = $request->getUserVar('temporaryFileId');
-				$temporaryFileDao = DAORegistry::getDAO('TemporaryFileDAO'); /* @var $temporaryFileDao TemporaryFileDAO */
-				$user = $request->getUser();
-				$temporaryFile = $temporaryFileDao->getTemporaryFile($temporaryFileId, $user->getId());
-				if (!$temporaryFile) {
-					$json = new JSONMessage(true, __('plugins.inportexport.native.uploadFile'));
-					header('Content-Type: application/json');
-					return $json->getString();
-				}
-				$temporaryFilePath = $temporaryFile->getFilePath();
+		parent::display($args, $request);
 
-				$deployment = new NativeImportExportDeployment($context, $user);
+		if ($this->isResultManaged) {
+			if ($this->result) {
+				return $this->result;
+			}
 
-				libxml_use_internal_errors(true);
-				$submissions = $this->importSubmissions(file_get_contents($temporaryFilePath), $deployment);
-				$templateMgr->assign('submissions', $submissions);
-				$validationErrors = array_filter(libxml_get_errors(), function($a) {
-					return $a->level == LIBXML_ERR_ERROR ||  $a->level == LIBXML_ERR_FATAL;
-				});
-				$templateMgr->assign('validationErrors', $validationErrors);
-				libxml_clear_errors();
+			return false;
+		}
 
-				// Are there any submissions import errors?
-				$submissionsErrors = $deployment->getProcessedObjectsErrors(ASSOC_TYPE_SUBMISSION);
-				if (!empty($submissionsErrors)) {
-					$templateMgr->assign('submissionsErrors', $submissionsErrors);
-				}
-				// Are there any submissions import warnings?
-				$submissionsWarnings = $deployment->getProcessedObjectsWarnings(ASSOC_TYPE_SUBMISSION);
-				if (!empty($submissionsWarnings)) {
-					$templateMgr->assign('submissionsWarnings', $submissionsWarnings);
-				}
-				// If there are any submissions or validataion errors
-				// delete imported submissions.
-				if (!empty($submissionsErrors) || !empty($validationErrors)) {
-					// remove all imported submissions
-					$deployment->removeImportedObjects(ASSOC_TYPE_SUBMISSION);
-				}
-				// Display the results
-				$json = new JSONMessage(true, $templateMgr->fetch($this->getTemplateResource('results.tpl')));
-				header('Content-Type: application/json');
-				return $json->getString();
-			case 'export':
-				$exportXml = $this->exportSubmissions(
-					(array) $request->getUserVar('selectedSubmissions'),
-					$request->getContext(),
-					$request->getUser()
-				);
-				import('lib.pkp.classes.file.FileManager');
-				$fileManager = new FileManager();
-				$exportFileName = $this->getExportFileName($this->getExportPath(), 'monographs', $context, '.xml');
-				$fileManager->writeFile($exportFileName, $exportXml);
-				$fileManager->downloadByPath($exportFileName);
-				$fileManager->deleteByPath($exportFileName);
-				break;
+		switch ($this->opType) {
 			default:
 				$dispatcher = $request->getDispatcher();
 				$dispatcher->handle404();
```

Comment:
Won't this always result in a call to `handle404`?

Reply:
Not if the display op is handled by the parent class (see line 31)

Reply:
Ah, I see!